lib/std/terminfo.zig

```diff
@@ -0,0 +1,1069 @@
+    .{ .short = "s1ds", .long = .{ .string = .set1_des_seq } },
+    .{ .short = "s2ds", .long = .{ .string = .set2_des_seq } },
+    .{ .short = "s3ds", .long = .{ .string = .set3_des_seq } },
+    .{ .short = "setab", .long = .{ .string = .set_a_background } },
+    .{ .short = "setaf", .long = .{ .string = .set_a_foreground } },
+    .{ .short = "setcolor", .long = .{ .string = .set_color_band } },
+    .{ .short = "smglr", .long = .{ .string = .set_lr_margin } },
+    .{ .short = "slines", .long = .{ .string = .set_page_length } },
+    .{ .short = "smgtb", .long = .{ .string = .set_tb_margin } },
+    .{ .short = "ehhlm", .long = .{ .string = .enter_horizontal_hl_mode } },
+    .{ .short = "elhlm", .long = .{ .string = .enter_left_hl_mode } },
+    .{ .short = "elohlm", .long = .{ .string = .enter_low_hl_mode } },
+    .{ .short = "erhlm", .long = .{ .string = .enter_right_hl_mode } },
+    .{ .short = "ethlm", .long = .{ .string = .enter_top_hl_mode } },
+    .{ .short = "evhlm", .long = .{ .string = .enter_vertical_hl_mode } },
+    .{ .short = "sgr1", .long = .{ .string = .set_a_attributes } },
+    .{ .short = "slength", .long = .{ .string = .set_pglen_inch } },
+};
+
+pub fn getCapsLookUpTable(allocator: *std.mem.Allocator) !std.StringHashMap(Terms) {
+    var data = std.StringHashMap(Terms).init(allocator);
+    for (dictionary) |term| 
+        _ = try data.put(term.short, term.long);
+    return data;
+}
+
+pub const Terminfo = struct {
+    names: std.StringHashMap(bool),
+    caps: std.StringHashMap([]const u8),
+
+    pub fn init(allocator: *std.mem.Allocator, terminfo_path: []const u8, cap_delimiter: []const u8) !Terminfo {
+        var names = std.StringHashMap(bool).init(allocator);
+        var caps = std.StringHashMap([]const u8).init(allocator);
+        _ = try caps.put("_name", "tui.terminfo capabilities");
+
+        var read_line = try std.fs.ReadLineIterator.init(terminfo_path);
+        defer read_line.deinit();
+        read_line.setFilter(struct {
+            fn filter(line: []u8) bool {
+                return !(std.mem.startsWith(u8, line, "#") or line.len == 0);
+            }
+        }.filter);
+        read_line.setMap(struct {
+            fn map(line: []u8) []const u8 {
+                return std.mem.trim(u8, line[0..], " \r\t");
+            }
+        }.map);
+
+        var buffer = try read_line.collect(allocator);
+        defer buffer.deinit();
+        var tokens = std.mem.tokenize(buffer.toSlice(), cap_delimiter);
+
+        while (tokens.next()) |token| {
+            const trimmed = std.mem.trimLeft(u8, token, " ");
+
+            if (std.mem.indexOf(u8, trimmed, "|") != null) {
+                var name_tokens = std.mem.tokenize(trimmed, "|");
+                while (name_tokens.next()) |name|
+                    _ = try names.put(name, true);
+            } else if (std.mem.indexOf(u8, trimmed, "=")) |equals| {
+                _ = try caps.put(trimmed[0..equals], trimmed[equals+1..]);
```

Comment:
Isn't `trimmed` a slice pointing into `buffer`? which is freed at end of this scope?

Reply:
I'm not following your question here based on what you have selected. Sorry for my ignorance. Can you explain to me what's going on here?

Reply:
calling `deinit` doesn't seem to free any memory, only close the file.